src/components/Articles.js

```diff
@@ -1,29 +1,36 @@
 import React from 'react'
+import ArticleCard from "./ArticleCard";
 import { useEffect, useState } from 'react'
-import { getArticles } from '../utils/api'
+import { getArticles, getTopicsApi } from '../utils/api'
 import { useParams, Link} from 'react-router-dom';
 
 
 const Articles = () => {
   const [articles, setArticles] = useState([]); 
   const { article_id } = useParams();
+  const { topic } = useParams();
+  console.log(topic)
   
   useEffect(() => {
     getArticles().then(articlesFromApi => {
       setArticles(articlesFromApi.articles)
     });
-  },[article_id]);
+    getTopicsApi(topic).then(topicFromArticle => {
+      setArticles(topicFromArticle)
+    });
+  },[topic]);
```

Comment:
When you have two different functions for getting all the articles and getting the articles for a certain topic, you will need to have some conditional logic. At the moment, when we click the articles button to see all the articles we get an error because getTopicsApi is being invoked with topic which will be undefined in this case.

It's also possible to rewrite your getArticles function to work for both getting all articles and getting all articles of a certain topic depending on if we pass in a topic 😄